remote/Dockerfile

```
@@ -47,6 +47,7 @@ RUN wget -qO /usr/share/keyrings/xpra.asc ${XPRA_REGISTRY}/xpra.asc && \
 
 COPY rc.xml /etc/xdg/openbox/rc.xml
 COPY menu.xml /etc/xdg/openbox/menu.xml
+RUN printf '\n%s\n' GTK_IM_MODULE=ibus >> /etc/environment
```

```suggestion
RUN echo -e 'GTK_IM_MODULE=ibus\n'  >> /etc/environment
```
I find that sorter, easier to read, and possibly edited (or extended in the future).
If this suggestion will not be applied, the leading `\n` should be removed, because `>>` is already adding a leading new line.

```bash
rm -f test && echo "1" > test && printf '%s\n' GTK_IM_MODULE=ibus >> test && cat test
```
gives us:

```text
1
GTK_IM_MODULE=ibus
```

---

The `>>` shell operator doesn't do anything to the file by itself. The behavior you described relies on the trailing newline added by the `echo 1` command.

> If -n is specified, the trailing newline  is suppressed.\
> https://man.archlinux.org/man/core/bash/bash.1.en#echo~2

```
$ echo -n 1 > test.txt
$ printf '%s\n' 2 >> test.txt
$ cat test.txt
12
$
```

A simple `echo foo >> bar` works in almost all cases because it's so common to have a trailing newline in text files, however we support using arbitrary base images, so I wanted to minimize the number of assumptions.

---

This point goes to you :-)